### DISCUSSION TITLE
thread 'main' panicked at 'assertion failed: `(left == right)` When including .add_plugins(DefaultPlugins) - Using PyCharm

### DISCUSSION THREAD

**Discussion by XcaliberRage** [bevyengine/bevy#2595] [Q&A]
Hi I'm very new to Bevy and was following along the first steps for the Snake Game when I ran into this problem.

This specifically happens due to me adding .add_plugins(DefaultPlugins) to the App:build() line.

> thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `-2005270523`,
 right: `0`: error on command list creation: 887a0005', C:\Users\UserName\.cargo\registry\src\github.com-1ecc6299db9ec823\gfx-backend-dx12-0.7.1\src\pool.rs:105:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aug 03 15:40:07.523 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.523 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.524 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.524 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.525 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.525 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.525 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.525 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.526 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.526 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.526 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.526 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.527 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.527 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.527 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 03 15:40:07.527 ERROR gpu_alloc::block: Memory block wasn't deallocated
error: process didn't exit successfully: `target\debug\BevyChessTutorial.exe RUST_BACKTRACE=full` (exit code: 101)

When I run cargo version I get:
> cargo 1.55.0-nightly (d21c22870 2021-07-26)

And for reference my code is:
> use bevy::prelude::*;

> fn main() {
>     App::build().add_plugins(DefaultPlugins).run();
> }

I have this all set up on PyCharm Community edition build #PC-212.4746.96
And I'm certain that both Rust and Bevy are fully up to date.

The usual result is the window will pop up, hang for some seconds before closing with the above displayed error message in the Terminal.

I've spent some time trying to find anyone else with similar issues and I really can't find anything that's helpful. So I apologise if this is a duplicate!

**Comment by MinerSebas** [comment]
This seems like a Graphics Problem.
What graphics card do you use?
Can you also try running a wgpu example: https://github.com/gfx-rs/wgpu/tree/master/wgpu/examples

**Reply by XcaliberRage** [reply]
I tried to run a few examples and I kept getting back that they didn't exist.

when I run `cargo run --example` I get `No examples available`

My GPU is Intel HD Graphics 4600

**Reply by bjorn3** [reply]
> when I run cargo run --example I get No examples available

You need to pick a specific example. Eg `cargo run --example hello-triangle`.

**Reply by XcaliberRage** [reply]
Sorry I wasn't clear, I tried that and got `no example target named 'hello-triangle'` repeat for multiple different options

**Reply by MinerSebas** [reply]
In which folder are you running cargo?

**Reply by XcaliberRage** [reply]
C:\Users\<UserName>\PycharmProjects\BevyChessTutorial

**Reply by MinerSebas** [reply]
You need to clone the Linked repository, to build its examples.

**Reply by XcaliberRage** [reply]
Apologies! I thought I understood they were meant to be automatically included.

I tried running it and after it built I got a brief window pop up and then this error:

> Using Intel(R) HD Graphics 4600 (Dx12)
[2021-08-03T15:01:40Z ERROR wgpu_hal::dx12::instance] ID3D12Device::RemoveDevice: Device removal has been triggered for the following reason (DXGI_ERROR_DEVICE_HUNG: The Device took an unreasonable amount of time to execute its comm
ands, or the hardware crashed/hung. As a result, the TDR (Timeout Detection and Recovery) mechanism has been triggered. The current Device Context was executing commands when the hang occurred. The application may want to respawn an
d fallback to less aggressive use of the display hardware). [ EXECUTION ERROR #232: DEVICE_REMOVAL_PROCESS_AT_FAULT]
error: process didn't exit successfully: `target\debug\examples\boids.exe` (exit code: 1)

Also I have to log off now and won't be able to test further until Thursday so don't worry about a lack of response until then!

**Comment by XcaliberRage** [comment]
In further testing, I tried running a wgpu-rs example (hello-triangles) and got a white winit window but no crash until I tried clicking in the window where I got this:

> thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `-2005270523`,
 right: `0`: error on command list creation: 887a0005', C:\Users\UserName\.cargo\git\checkouts\gfx-e86e7f3ebdbc4218\27a1dae\src\backend\dx12\src\pool.rs:105:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2021-08-05T11:33:53Z ERROR wgpu_core::command::allocator] Some command buffers are still recorded, only tracking 19 / 20
error: process didn't exit successfully: `target\debug\examples\hello-triangle.exe` (exit code: 101)
PS C:\Users\UserName\PycharmProjects\GitHub\wgpu-rs>
